### ISSUE TITLE
training issue with HamGNN

### ISSUE THREAD

**Issue by gaoxuew** [QuantumLab-ZY/HamGNN#32]
Dear Dr. Zhong,
I have been trying to test HamGNN for a system of interest, but encountered issues to reduce the training error. See the following correlation plot. I would really appreciate if you have any suggestions for improving this. I have uploaded the graph inputs at: https://zenodo.org/records/13941814. I also attached the config.yaml that I used below. Thank you.
Best,
Gaoxue

<img width="529" alt="Screenshot 2024-10-16 at 11 13 18 AM" src="https://github.com/user-attachments/assets/53807dea-9c65-4983-880a-8dd9f45b8860">

config.yaml:
dataset_params:
  batch_size: 1
  split_file: null
  test_ratio: 0.1
  train_ratio: 0.8
  val_ratio: 0.1
  graph_data_path: ./ # Directory where graph_data.npz is located

losses_metrics:
  losses:
  - loss_weight: 1.0
    metric: mae
    prediction: hamiltonian
    target: hamiltonian

  metrics:
  - metric: mae
    prediction: hamiltonian
    target: hamiltonian

# Generally, the optim_params module only needs to set the initial learning rate (lr)
optim_params:
  lr: 0.01
  lr_decay: 0.5
  lr_patience: 5
  gradient_clip_val: 0.0
  max_epochs: 3000
  min_epochs: 100
  stop_patience: 30

output_nets:
  output_module: HamGNN_out
  HamGNN_out:
    ham_only: true # true: Only the Hamiltonian H is computed; 'false': Fit both H and S
    ham_type: openmx # openmx: fit openmx Hamiltonian; abacus: fit abacus Hamiltonian
    nao_max: 19 # The maximum number of atomic orbitals in the data set, which can be 14, 19 or 27
    add_H0: true # Generally true, the complete Hamiltonian is predicted as the sum of H_scf plus H_nonscf (H0)
    symmetrize: true # if set to true, the Hermitian symmetry constraint is imposed on the Hamiltonian
    calculate_band_energy: false # Whether to calculate the energy bands to train the model
    num_k: 5 # When calculating the energy bands, the number of K points to use
    band_num_control: null # `dict`: controls how many orbitals are considered for each atom in energy bands; `int`: [vbm-num, vbm+num]; `null`: all bands
    k_path: null # `auto`: Automatically determine the k-point path; `null`: random k-point path; `list`: list of k-point paths provided by the user
    soc_switch: false # if true, fit the SOC Hamiltonian
    nonlinearity_type: norm # norm or gate

profiler_params:
  progress_bar_refresh_rat: 1
  train_dir: ./ #The folder for saving training information and prediction results. This directory can be read by tensorboard to monitor the training process.

representation_nets:
  # Network parameters usually do not need to be changed.
  HamGNN_pre:
    cutoff: 26.0
    resnet: True
    cutoff_func: cos
    edge_sh_normalization: component
    edge_sh_normalize: true
    ######## Irreps set 1 (crystal): ################
    feature_irreps_hidden: 64x0e+32x1o+16x1e+8x2o+20x2e+8x3o+4x3e+4x4e
    irreps_edge_output: 64x0e+32x1o+16x1e+8x2o+20x2e+8x3o+4x3e+4x4e
    irreps_edge_sh: 0e + 1o + 2e + 3o + 4e + 5o
    irreps_node_features: 64x0e+32x1o+16x1e+8x2o+20x2e+8x3o+4x3e+4x4e
    irreps_node_output: 64x0e+32x1o+16x1e+8x2o+20x2e+8x3o+4x3e+4x4e
    irreps_triplet_output: 64x0e+32x1o+16x1e+8x2o+20x2e+8x3o+4x3e+4x4e
    invariant_layers: 3
    invariant_neurons: 64
    num_interaction_layers: 3
    num_radial: 64
    num_spherical: 8
    num_types: 60
    export_triplet: false
    rbf_func: bessel
    set_features: true
    add_edge_tp: False
    irreps_node_prev: 64x0e+20x1o+8x1e+8x2o+20x2e+8x3o+4x3e+4x4e
    num_node_attr_feas: 64

setup:
  GNN_Net: HamGNN_pre
  accelerator: null
  ignore_warnings: true
  checkpoint_path: ./ # Path to the model weights file
  load_from_checkpoint: false
  resume: false
  num_gpus: null # null: use cpu; [i]: use the ith GPU device
  precision: 32
  property: hamiltonian
  stage: fit # fit: training; test: inference

**Comment by QuantumLab-ZY** [comment]
Hi @gaoxuew, Can you show me the openmx .dat file used in your training set? It seems that the problem is caused by the mismatch between the atomic orbital basis of your Te element and that in HamGNN.

**Comment by QuantumLab-ZY** [comment]
Hi @gaoxuew, you can add `55:[0,1,2,3,4,5,6,7,8,9,10,11 ,12 ,13 ,14 ,15 ,16 ,17 ,18], # Te` to the dictionary `self.basis_def`(Line: 1031) in [net.py](https://github.com/QuantumLab-ZY/HamGNN/edit/main/HamGNN/models/HamGNN/net.py).

**Comment by gaoxuew** [comment]
Thank you. That’s helpful. I have another question on getting charge density as the one you showed in your paper on silicon with defect. May I know the package/scrips that you used to process the predicted Hamiltonian to get other materials properties?
